sdlib/d/gc/extent.d

```diff
@@ -233,14 +305,65 @@ public:
 
 		bits += (1UL << 48);
 		slabData.clearBit(index);
+
+		// Clear meta flags, if they exist
+		if (slabSlots <= 256)
+			apFlags.clearBit(index);
+
+		if (slabSlots <= 128)
+			finFlags.clearBit(index);
+	}
+
+	@property
+	bool isAppendable(uint index) {
+		assert(isSlab(), "isAppendable(index) accessed on non slab!");
+
+		// 512-slot classes are never appendable
+		if (slabSlots == 512)
+			return false;
+
+		return apFlags.valueAt(index);
+	}
+
+	@property
+	bool isFinalizable(uint index) {
```

Comment:
I highly doubt this is the API we want. In what context will we want to ask the question in this manner? When collecting, we'll probably go over the whole slab either asking the question or not depending on the sizeclass right out of the bat. That question would be easy to answer if the code actually did something.

Reply:
`isFinalizable` is intended to be used when destroying a slab allocation (in which case the finalizer is expected to be present in the tail of the block, and is invoked.) `isAppendable` is intended to be used when carrying out an operation which requires appendability (or invoking the predicated which determines whether such operations may be used) on a slab allocation.

Reply:
Please note that it is impossible to answer the question of e.g. `isFinalizable` simply by looking at a slab's size class -- a slab in that class may nonetheless not contain a finalizer. Similarly for `isAppendable`. I did not exclude these classes from containing "non-special" allocs (i.e. without a finalizer or length appendix.)